rts/lra/test/arquillian-extension/src/main/java/io/narayana/lra/arquillian/spi/NarayanaLRARecovery.java

```diff
@@ -57,7 +57,14 @@
 
     @Override
     public void waitForCallbacks(URI lraId) {
-        // no action needed
+        // for tests that do not involve timeouts no action is needed,
+        // but there are races on time sensitive tests (depending upon how busy the test system is).
+        // To handle these cases introduce an arbitrary 1000 ms delay (note that since LRA provides
+        // no real-time guarantees, introducing the delay is a valid approach):
+        try {
+            Thread.sleep(1000);
```

Comment:
I read the explanation https://issues.redhat.com/browse/JBTM-3513?focusedCommentId=18965208#comment-18965208 and I do agree with that. But I don't think it's good to add this arbitrary long sleep time here.
For other environment the `1000` can be not enough, on many other it means to slow down the test execution when not needed.

I was trying to address the same issue with adding the annotation extension which consider `lra.test.timeout.factor` and applies it to the `@LRA` annotation definition.
https://github.com/jbosstm/narayana/blob/master/rts/lra/test/arquillian-extension/src/main/java/io/narayana/lra/arquillian/spi/LRAAnnotationAdjuster.java#L166
-> https://github.com/jbosstm/narayana/blob/master/rts/lra/test/tck/pom.xml#L54

I hope that's the reason why we don't experience the `timeLimit` failure at CI runs.

Additional note:
I can see that a trouble could be with `-Parq` that the timeout is not passed to the WildFly runtime (https://github.com/jbosstm/narayana/blob/master/rts/lra/test/tck/src/test/resources/arquillian-wildfly.xml).

To me this adjustment could not be needed, maybe.

Reply:
@ochaloup 
> To me this adjustment could not be needed, maybe.

The comment that you referenced (the one in JBTM-3513) already stated why I didn't use `lra.test.timeout.factor`.